api/v1alpha1/clusterinstance_types.go

```diff
@@ -581,6 +582,18 @@ type ClusterInstanceStatus struct {
 	// +optional
 	DeploymentConditions []hivev1.ClusterDeploymentCondition `json:"deploymentConditions,omitempty"`
 
+	// Reference to the associated HostedCluster resource for HCP clusters.
+	// +optional
+	HostedClusterRef *corev1.LocalObjectReference `json:"hostedClusterRef,omitempty"`
+
+	// HostedCluster status for HCP clusters - vendored from hypershift API.
+	// +optional
+	HostedClusterStatus *hypershiftv1beta1.HostedClusterStatus `json:"hostedClusterStatus,omitempty"`
```

Comment:
Both Image based install and assisted install clusters roll up status under a generic set of conditions defined by Hive. It would be good for HCP clusters to roll up under the same set of status conditions rather than expanding the status section per cluster type. Ideally HostedControlPlane would use the same set of conditions, if not we would have to reinterpret here.